src/compiler/crystal/syntax/ast.cr

```diff
@@ -214,8 +214,8 @@ module Crystal
     def initialize(@value : String, @kind = :i32)
     end
 
-    def initialize(value : Number, @kind = :i32)
-      @value = value.to_s
+    def self.new(value : Number)
```

Comment:
Why not continue to accept `.new(Number, kind: Symbol)`, but have `.new(Number)` take the type of the number. This way you don't need the `to_s` additions above (they all have explicit kind anyway).

Reply:
I don't think this would be that useful. It's a simple API now: If the constructor receives a `Number`, it deduces `kind` accordingly. Otherwise you need to specify `value` (as string) and `kind` explicitly.